### DISCUSSION TITLE
[13] Dinder - Tests

### DISCUSSION THREAD

**Discussion by ghost** [dhbw-ka-tinf22b5-dinder/Dinder#20] [Projektblog]
# Tests

## Update
We added an explanation for the total test coverage in 5.2:
![TestPlan](https://github.com/dhbw-ka-tinf22b5-dinder/Dinder-SRS/blob/main/testplan.md)

# Further changes
## Frontend
- Migration from React and Vite to Next.js has been finished
## Backend
- Final implementation of Supabase Storage.
- Added endpoint to insert an advertisement into the database
- Also added an endpoint to load the image out of Supabase
- Implemented a basic test and included that into the pipeline
## Tests
- Research for automated API-tests in GitHub Actions/SonarCloud
- Research for tests with mocked beans in Spring Boot

**Comment by Fefke** [comment]
Abgerundeter und kompakter Blogbeitrag
und respekt vor eurem ausgereiften Testplan,
auf den ihr weiter iteriert. 
Besser kann man's kaum machen 🫶

Ich frage mich nur, wieso ihr in den letzten beiden Wochen von React on Vite nach Next.js gewechselt seid. Habt ihr vor euren Backend-Controller mit zu migrieren oder verwendet ihr Next.js strikt für das SSR-Frontend?

Liebe Grüße - IncidArch ⛑️

**Comment by Puggingtons** [comment]
Grüßle,
find ich gut, dass ihr den mutigen Schritt des Migrierens auf ein anderes Framework gegangen seid.
Next.js ist ja ein Full-Stack-Framework. Ist bei euch noch der Plan, auch das Backend noch umzuziehen?
